orttraining/orttraining/training_api/include/module.h

```diff
@@ -86,10 +86,16 @@ struct Module {
   Status GetStateDict(ModuleCheckpointState& module_checkpoint_states);
 
   // Returns the output count for training graph
-  size_t GetTrainModeOutputCount() const noexcept;
+  size_t GetTrainModelOutputCount() const noexcept;
 
   // Returns the output count for eval graph
-  size_t GetEvalModeOutputCount() const noexcept;
+  size_t GetEvalModelOutputCount() const noexcept;
+
+  // Returns the output names for train graph
+  void GetTrainModelOutputName(size_t& index, std::string& name) const;
```

Comment:
reference for `size_t` not needed.

Why return void? Can we return the [string instead](https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs)?